### PR TITLE
Introduce docker version check into scope script, error on <1.5.0

### LINKED ISSUE #548: i can't launch scope
hi, 

i try launch this project but i catch an error. the message is 

flag provided but not defined: --pid

my docker version is Docker version 1.0.1, build 990021a

thx in advance.

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #548

### CODE REVIEW

File: scope [hunk 1]
```diff
@@ -29,6 +29,34 @@ WEAVESCOPE_DOCKER_ARGS=${WEAVESCOPE_DOCKER_ARGS:-}
 COMMAND=$1
 shift 1
 
+# - The image embeds the weave script & Docker 1.3.1 client
+# - Docker versions prior to 1.5.0 do not support --pid=host
+MIN_DOCKER_VERSION=1.5.0
+
+check_docker_version() {
+    if ! DOCKER_VERSION=$(docker -v | sed -n -e 's|^Docker version \([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\).*|\1|p') || [ -z "$DOCKER_VERSION" ] ; then
```
Comment (paulbellamy) [inline scope]: Could also do this with: `docker -v | grep -oE "[0-9]+\.[0-9]+\.[0-9]+"`

File: scope [hunk 2]
```diff
@@ -29,6 +29,34 @@ WEAVESCOPE_DOCKER_ARGS=${WEAVESCOPE_DOCKER_ARGS:-}
...
+
+    MIN_DOCKER_VERSION_MAJOR=$(echo "$MIN_DOCKER_VERSION" | cut -d. -f 1)
+    MIN_DOCKER_VERSION_MINOR=$(echo "$MIN_DOCKER_VERSION" | cut -d. -f 2)
+    MIN_DOCKER_VERSION_PATCH=$(echo "$MIN_DOCKER_VERSION" | cut -d. -f 3)
+
+    if [ \( "$DOCKER_VERSION_MAJOR" -lt "$MIN_DOCKER_VERSION_MAJOR" \) -o \
+        \( "$DOCKER_VERSION_MAJOR" -eq "$MIN_DOCKER_VERSION_MAJOR" -a \
+        \( "$DOCKER_VERSION_MINOR" -lt "$MIN_DOCKER_VERSION_MINOR" -o \
+        \( "$DOCKER_VERSION_MINOR" -eq "$MIN_DOCKER_VERSION_MINOR" -a \
+        \( "$DOCKER_VERSION_PATCH" -lt "$MIN_DOCKER_VERSION_PATCH" \) \) \) \) ] ; then
```
Comment (paulbellamy) [inline scope]: This is awful, but short of using gnu sort...

File: scope (hunk 1 above)
Comment (tomwilkie) [inline scope]: This is just copied from the weave script; I'll see if we can apply the same simplification there.

**Comment by tomwilkie** [comment]
@paulbellamy PTAL

**Comment by paulbellamy** [comment]
LGTM

**Pull request merged**